File: tl/src/main/java/com/github/badoualy/telegram/tl/api/TLAbsUserProfilePhoto.java
package com.github.badoualy.telegram.tl.api;

import com.github.badoualy.telegram.tl.core.TLObject;

/**
 * Abstraction level for the following constructors:
 * <ul>
 * <li>{@link TLUserProfilePhoto}: userProfilePhoto#d559d8c8</li>
 * <li>{@link TLUserProfilePhotoEmpty}: userProfilePhotoEmpty#4f11bae1</li>
 * </ul>
 *
 * @author Yannick Badoual yann.badoual@gmail.com
 * @see <a href="http://github.com/badoualy/kotlogram">http://github.com/badoualy/kotlogram</a>
 */
public abstract class TLAbsUserProfilePhoto extends TLObject {

    public TLAbsUserProfilePhoto() {
    }

    public abstract boolean isEmpty();

    public abstract boolean isNotEmpty();

    public TLUserProfilePhoto getAsUserProfilePhoto() {
        return null;
    }
}

File: tl/src/main/kotlin/com/github/badoualy/telegram/tl/api/TLAbsUserProfilePhoto.kt
package com.github.badoualy.telegram.tl.api

import com.github.badoualy.telegram.tl.TLObjectUtils.SIZE_BOOLEAN
import com.github.badoualy.telegram.tl.TLObjectUtils.SIZE_CONSTRUCTOR_ID
import com.github.badoualy.telegram.tl.TLObjectUtils.SIZE_DOUBLE
import com.github.badoualy.telegram.tl.TLObjectUtils.SIZE_INT32
import com.github.badoualy.telegram.tl.TLObjectUtils.SIZE_INT64
import com.github.badoualy.telegram.tl.TLObjectUtils.computeTLBytesSerializedSize
import com.github.badoualy.telegram.tl.TLObjectUtils.computeTLStringSerializedSize
import com.github.badoualy.telegram.tl.core.TLObject
import kotlin.Boolean

/**
 * Abstraction level for the following constructors:
 * * [userProfilePhoto#d559d8c8][TLUserProfilePhoto]
 * * [userProfilePhotoEmpty#4f11bae1][TLUserProfilePhotoEmpty]
 *
 * @author Yannick Badoual yann.badoual@gmail.com
 * @see <a href="http://github.com/badoualy/kotlogram">http://github.com/badoualy/kotlogram</a>
 */
abstract class TLAbsUserProfilePhoto : TLObject() {
    fun isEmpty(): Boolean = this is TLUserProfilePhotoEmpty

    fun isNotEmpty(): Boolean = this is TLUserProfilePhoto

    fun asUserProfilePhoto(): TLUserProfilePhoto? = this as? TLUserProfilePhoto
}
